src/Features/CSharp/Portable/CodeMapper/CSharpMapCodeService.ReplaceMapperHelper.cs

```diff
@@ -0,0 +1,292 @@
+
+            // remove removeLeadingTrivia from matching node
+            var adjustedMatchingNode = matchingNode
+                .WithoutLeadingTrivia()
+                .WithLeadingTrivia(leadingTrivia.Except(removeLeadingTrivia));
+            var newLength = adjustedMatchingNode.FullSpan.Length;
+            var diff = originalLength - newLength;
+            textSpan = new TextSpan(textSpan.Value.Start + diff, newLength);
+            return (adjustedMatchingNode, textSpan.Value);
+        }
+
+        private static (SyntaxNode, TextSpan) RemoveLeadingCommentsTrivia(SyntaxNode matchingNode, TextSpan? textSpan = null)
+        {
+            // When a textSpan is not provided, we assume that the node is the original node.
+            if (textSpan is null)
+            {
+                textSpan = matchingNode.FullSpan;
+            }
+
+            var originalLength = textSpan.Value.Length;
+            var leadingTrivia = matchingNode.GetLeadingTrivia();
+            var removeLeadingTrivia = new List<SyntaxTrivia>();
+            foreach (var trivia in leadingTrivia)
+            {
+                if (IsCommentOrXmlDocTrivia(trivia))
+                {
+                    removeLeadingTrivia.Add(trivia);
+                }
+            }
+
+            // remove removeLeadingTrivia from matching node
+            matchingNode = matchingNode
+                .WithoutLeadingTrivia()
+                .WithLeadingTrivia(leadingTrivia.Except(removeLeadingTrivia));
+            var newLength = matchingNode.FullSpan.Length;
+            var diff = originalLength - newLength;
+            textSpan = new TextSpan(textSpan.Value.Start + diff, newLength);
+            return (matchingNode, textSpan.Value);
+        }
+
+        private static bool IsCommentOrXmlDocTrivia(SyntaxTrivia t)
+        {
+            return t.IsKind(SyntaxKind.DocumentationCommentExteriorTrivia)
+                    || t.IsKind(SyntaxKind.MultiLineCommentTrivia)
+                    || t.IsKind(SyntaxKind.DocumentationCommentExteriorTrivia)
+                    || t.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia)
+                    || t.IsKind(SyntaxKind.MultiLineDocumentationCommentTrivia)
+                    || t.IsKind(SyntaxKind.SingleLineCommentTrivia);
+        }
+
+        private static bool ShouldSkipAttributes(AttributeSyntax[] insertionAttributes, AttributeSyntax[] targetAttributes)
+        {
+            if (insertionAttributes.Length == 0)
+            {
+                return true;
+            }
+
+            if (insertionAttributes.Length == targetAttributes.Length)
+            {
+                foreach (var insertionAttribute in insertionAttributes)
+                {
+                    // If we find at least one attribute that doesn't exists on target, we'll take it as we need to
+                    // replace those attributes.
+                    if (!targetAttributes.Any(a => a.ToString() == insertionAttribute.ToString()))
```

Comment:
these tostring calls worry me a *lot*.  even a single spacing different anywhere would make thse not match.

Reply:
IMO that's ok.
We want to be super certain that a node matches exactly an existing node in the code before declaring it as 'replaceable' by the mapper.
The reasoning behind this is that, we have another 'Default' mapper that will match nodes based on text differences, but that one will only be calculated when the C# mapper determines there's nothing to replace. 
So when the C# mapper determines that something is repleaceable, it means we are most certain about it, and the default mapper will not be taken into consideration.

The scenario you mention with space differences should be covered by the default mapper.